src/Microsoft.Data/PrimitiveDataFrameColumn.cs

```diff
@@ -477,7 +493,12 @@ public override DataFrameColumn Clip<U>(U lower, U upper, bool inPlace = false)
                 throw new ArgumentException(Strings.MismatchedValueType + typeof(T).ToString(), nameof(U));
         }
 
-        public override DataFrameColumn Filter<U>(U lower, U upper)
+        protected override DataFrameColumn ClipImplementation<U>(U lower, U upper, bool inPlace)
+        {
+            return Clip(lower, upper, inPlace);
+        }
+
+        public new PrimitiveDataFrameColumn<T> Filter<U>(U lower, U upper)
```

Comment:
Could be 

```C#
public new PrimitiveDataFrameColumn<T> Filter(T lower, T upper)
```

Reply:
Same explanation as the `FillNulls` comment. If `U` can be converted to `T`, I'd like to let that through.

Reply:
User's can cast, or they can use implicit conversions, if they exist. Our strongly-typed APIs shouldn't be in the business of doing type conversion.